core-experimental/include/iDynTree/Core/VectorFixSize.h

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (C) 2015 Fondazione Istituto Italiano di Tecnologia
+ * Authors: Silvio Traversaro
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ *
+ */
+
+#ifndef IDYNTREE_VECTOR_FIX_SIZE_H
+#define IDYNTREE_VECTOR_FIX_SIZE_H
+
+#include <iDynTree/Core/IVector.h>
+#include <iDynTree/Core/Utils.h>
+#include <string>
+#include <sstream>
+#include <cstring>
+
+namespace iDynTree
+{
+    /**
+     * Class providing a simple vector of N elements.
+     *  The size of the vector is known at compile time,
+     *  and it enables to avoid using dynamic memory allocation.
+     *
+     * \ingroup iDynTreeCore
+     */
+    template<unsigned int VecSize> class VectorFixSize: public IVector
+    {
+    protected:
+        /**
+         * Storage for the VectorDynSize
+         *
+         * Array of 6 doubles.
+         */
+        double m_data[VecSize];
+
+    public:
+        /**
+         * Default constructor: initialize the elements of the array to zero.
+         */
+        VectorFixSize();
+
+        /**
+         * Constructor from a C-style array.
+         *
+         * Print an error an build a vector full of zeros if in_size is not size().
+         */
+        VectorFixSize(const double * in_data, const unsigned int in_size);
+
+        /**
+         * Denstructor
+         *
+         */
+        virtual ~VectorFixSize();
+
+        /**
+         * @name Vector interface methods.
+         * Methods exposing a vector-like interface to VectorFixSize.
+         */
+        ///@{
+        double operator()(const unsigned int index) const;
```

Comment:
The documentation for this methods should be inherited from the `IVector` interface.